tests/arrays.rs

```diff
@@ -74,6 +74,28 @@ fn simple_array_indexing() {
     assert_eq!(metadata.count, 3);
 }
 
+#[test]
+fn array_iter() {
+    #[derive(Miniconf, Default, Clone, Copy, Debug, PartialEq)]
+    struct Inner {
+        b: u8,
+    }
+
+    #[derive(Miniconf, Default)]
+    struct S {
+        #[miniconf(defer)]
+        a: miniconf::Array<miniconf::Array<Inner, 2>, 2>,
+    }
+
+    let mut s = S::default();
+
+    for _i in s.a.into_iter().flatten() {}
+
+    for _i in s.a.iter().flatten() {}
+
+    for _i in s.a.iter_mut().flatten() {}
```

Comment:
Can we make these actually iterate across something to verify iteration works? Seems odd to me to do nothing here. Maybe even just an `iter().collect()` and compare to a known vec?